bundles/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/internal/discovery/NukiBridgeDiscoveryService.java

```diff
@@ -0,0 +1,168 @@
+import org.openhab.binding.nuki.internal.constants.NukiLinkBuilder;
+import org.openhab.binding.nuki.internal.dto.BridgeApiAuthDto;
+import org.openhab.binding.nuki.internal.dto.WebApiBridgeDiscoveryDto;
+import org.openhab.binding.nuki.internal.dto.WebApiBridgeDto;
+import org.openhab.core.common.ThreadPoolManager;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.id.InstanceUUID;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * Discovery service which uses Nuki Web API to discover all bridges on same network
+ * and uses authentication API to obtain access token.
+ *
+ * @author Jan Vybíral - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = DiscoveryService.class, configurationPid = "discovery." + NukiBindingConstants.BINDING_ID)
+public class NukiBridgeDiscoveryService extends AbstractDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(NukiBridgeDiscoveryService.class);
+
+    private final HttpClient httpClient;
+    private final Gson gson = new Gson();
+    private final ScheduledExecutorService threadPool = ThreadPoolManager.getScheduledPool("discovery");
+
+    @Activate
+    public NukiBridgeDiscoveryService(@Reference final HttpClientFactory httpClientFactory) {
+        super(Collections.singleton(NukiBindingConstants.THING_TYPE_BRIDGE), 30, false);
+        this.httpClient = httpClientFactory.getCommonHttpClient();
+    }
+
+    @Override
+    protected void startScan() {
+        logger.info("Bridge discovery started");
+
+        try {
+            ContentResponse response = this.httpClient.GET(NukiLinkBuilder.URI_BRIDGE_DISCOVERY);
+            if (response.getStatus() == HttpStatus.OK_200) {
+                String responseString = response.getContentAsString();
+                WebApiBridgeDiscoveryDto discoveryResult = gson.fromJson(responseString,
+                        WebApiBridgeDiscoveryDto.class);
+                if (discoveryResult == null) {
+                    logger.error("Bridge discovery failed - API returned invalid body {}", responseString);
+                } else if (discoveryResult.getErrorCode() == 0) {
+                    discoverBridges(discoveryResult);
+                } else {
+                    logger.error("Bridge discovery failed - API returned error code '{}': {}",
+                            discoveryResult.getErrorCode(), responseString);
```

Comment:
Bindings should only log to error if something severe happened, like the detection of a bug in your code. This could be debug. See this link for a description of the log levels: https://www.openhab.org/docs/developer/guidelines.html#f-logging

Please check all.